File: sdks/csharp/sdk/src/PogodocApi/Templates/Requests/SaveCreatedTemplateRequest.cs
using System.Text.Json.Serialization;
using PogodocApi.Core;

namespace PogodocApi;

[Serializable]
public record SaveCreatedTemplateRequest
{
    [JsonPropertyName("templateInfo")]
    public required SaveCreatedTemplateRequestTemplateInfo TemplateInfo { get; set; }

    [JsonPropertyName("previewIds")]
    public required SaveCreatedTemplateRequestPreviewIds PreviewIds { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/client/src/Templates/Requests/SaveCreatedTemplateRequest.php
<?php

namespace Pogodoc\Templates\Requests;

use Pogodoc\Core\Json\JsonSerializableType;
use Pogodoc\Templates\Types\SaveCreatedTemplateRequestTemplateInfo;
use Pogodoc\Core\Json\JsonProperty;
use Pogodoc\Templates\Types\SaveCreatedTemplateRequestPreviewIds;

class SaveCreatedTemplateRequest extends JsonSerializableType
{
    /**
     * @var SaveCreatedTemplateRequestTemplateInfo $templateInfo
     */
    #[JsonProperty('templateInfo')]
    public SaveCreatedTemplateRequestTemplateInfo $templateInfo;

    /**
     * @var SaveCreatedTemplateRequestPreviewIds $previewIds
     */
    #[JsonProperty('previewIds')]
    public SaveCreatedTemplateRequestPreviewIds $previewIds;

    /**
     * @param array{
     *   templateInfo: SaveCreatedTemplateRequestTemplateInfo,
     *   previewIds: SaveCreatedTemplateRequestPreviewIds,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->templateInfo = $values['templateInfo'];
        $this->previewIds = $values['previewIds'];
    }
}
